app/src/org/commcare/CommCareApplication.java

```diff
@@ -755,6 +761,30 @@ public void onServiceDisconnected(ComponentName className) {
         sessionServiceIsBinding = true;
     }
 
+    // Hand off an app update task to the Android WorkManager
+    private void scheduleAppUpdate() {
+        Constraints constraints = new Constraints.Builder()
+                .setRequiredNetworkType(NetworkType.CONNECTED)
+                .setRequiresBatteryNotLow(true)
+                .build();
+
+        PeriodicWorkRequest updateRequest =
+                new PeriodicWorkRequest.Builder(UpdateWorker.class, PERIODICITY_FOR_UPDATE_IN_HOURS, TimeUnit.HOURS)

```

Comment:
It looks like this more or less takes us to a context where we retry auto update every 2 hours, right?

Server load-wise I think that's quite often if we're presuming that we tried to contact the server, succeeded, and saw there was no update.

Reply:
So the internal `UpdateWorker` [check here ](https://github.com/dimagi/commcare-android/blob/autoUpdateScheduling/app/src/org/commcare/update/UpdateWorker.kt#L63) will still respect the auto update frequency set for the app which should take care of server getting hit too often.

Reply:
@ctsims  Wondering if we should still respect the update frequency set in app properties for triggering the background update  worker. We might derive more value in terms of update uptake from ignoring this flag and instead have a more liberal frequency for update check that doesn't cause the server load to spike too much.